src/commands/source/download.rs

```diff
@@ -314,5 +314,8 @@ pub async fn download(
     }
 
     debug!("r = {:?}", r);
+
+    super::verify(matches, config, repo, progressbars).await?;
```

Comment:
I missed this in previous reviews but we shouldn't use the function that implements the CLI here as it is super hacky and error prone - it only works because the Clap CLI arguments have the same names but it also does a lot of work twice.

Looking at the code we should be able to call `verify_impl` directly and get the desired result. It looks like you'll have to copy the packages from `r` though to have it available here. But looking at the `flat_map()` above it actually seems best to move this into that code (after the `perform_download()` (must be careful with the locking logic though)). And if you want to do it perfectly then it should even be possible to modify `perform_download()` instead to do the check there and delete the file if there was a checksum mismatch (IMO that is the desired behavior as a build command would happily use a source where the verification failed as it only checks if the file is present, IIRC).